WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewHostingController.swift

```diff
@@ -302,6 +304,47 @@ private extension DashboardViewHostingController {
     }
 }
 
+// MARK: Google Ads campaigns
+private extension DashboardViewHostingController {
+    func configureGoogleAdsCard() {
+        rootView.onCreateNewGoogleAdsCampaign = { [weak self] in
+            self?.startGoogleAdsCampaignCoordinator(forCampaignCreation: true)
+        }
+
+        rootView.onShowAllGoogleAdsCampaigns = { [weak self] in
+            self?.startGoogleAdsCampaignCoordinator(forCampaignCreation: false)
+        }
+    }
+
+    func startGoogleAdsCampaignCoordinator(forCampaignCreation: Bool) {
+        guard let site = viewModel.stores.sessionManager.defaultSite,
+              let navigationController else {
+            return
+        }
+
+        let shouldAuthenticateAdminPage = {
+            /// If the site is self-hosted and user is authenticated with WPCom,
+            /// `AuthenticatedWebView` will attempt to authenticate and redirect to the admin page and fails.
+            /// This should be prevented 💀⛔️
+            guard site.isWordPressComStore || viewModel.stores.isAuthenticatedWithoutWPCom else {
+                return false
+            }
+            return true
+        }()
```

Comment:
Opinion: To me, this logic looks like something that can be moved to `GoogleAdsCampaignCoordinator`. Dashboard doesn't have to know how to authenticate the Google ads web view.

Reply:
Since this logic is also used in hub menu, I added an extension method in stores manager to reuse it in both places. Changes were made in 759327d29579eac730cd6e53990a458057468c57 and 14d8aff78c99e3ce1efc8c9667d2bfec78dc6a54.